packages/medusa/src/services/shipping-option.ts

```diff
@@ -378,7 +378,9 @@ class ShippingOptionService extends TransactionBaseService {
       )
     }
 
-    const amount = option.includes_tax ? cart.total! : cart.subtotal!
+    const amount = option.includes_tax
```

Comment:
Question: was that the culprit?

Reply:
Validating against subtotal instead of total is a disadvantage for merchants. Consider the following example:

**Cart**
```
{
   ...
   subtotal: 16000
   total: 8000
   discount_total: 8000
}
```

**Shipping option**
```
{
   name: "Free shipping",
   requirements: [{ type: "min_subtotal", amount: 10000 }]
   price: 0
}
```

In this context, the customer would be able to add the free shipping option to their cart, because their subtotal is larger than the defined requirement, but they will actually only pay 8000 due to the applied discount. From what I can understand, the common approach is to determine the validity of shipping options after the total has been discounted. But then again, there's no silver bullet.

Both approaches are valid, they just cater to each party involved in the sale. Therefore, one could argue, that this should be configurable from a merchant point of view. 

What do you think?

Reply:
Yeah I see the point, so in the majority of the cases, the options should be validated against the total to include everything, discount and taxes included. Got it

Reply:
In the past, we've intentionally used subtotal, so this is actually not a bug, but rather something I believe should be treated differently. Which is why, I'd love a couple of opinions on it.

Reply:
If this is something that is applicable to the entire store and do not lix the cases, then it could be globally configurable. Otherwise It would be nice to be configurable from the end point. The user could send an additional options to choose the "strategy" to either use one or the other